doc/Security.xml

```diff
@@ -947,13 +995,311 @@ Added certificate-based client authentication</revremark>
+      <para>The JWT payload shall include the following standard claims:</para>
+      <para>
+        <itemizedlist>
+          <listitem>
+            <para><emphasis>iss</emphasis>: Issuer, the URI of the server that generated the JWT.</para>
+          </listitem>
+          <listitem>
+            <para><emphasis>aud</emphasis>: Audience, a list of strings representing the recipients that the JWT is intended for. Each principal intended to process the JWT shall identify itself with a value in the audience claim. If the principal processing the claim does not identify itself with a value in the audience claim when this claim is present, then the JWT shall be rejected.</para>
+          </listitem>
+          <listitem>
+            <para><emphasis>exp</emphasis>: Expiration Time.</para>
+          </listitem>
+          <listitem>
+            <para><emphasis>nbf</emphasis>: Not before.</para>
+          </listitem>
+        </itemizedlist>
+      </para>
+      <para>For the <emphasis>exp</emphasis>, <emphasis>nbf</emphasis> claims, a device shall reject a token when the current time is not within the range of claims nbf and exp . It shall reject a token when at least one of the claims is missing.</para>
+      <para>The JWT payload shall include the <emphasis>roles</emphasis> claim, as defined within [RFC 7643]:</para>
+      <para>
+        <itemizedlist>
+          <listitem>
+            <para><emphasis>roles</emphasis>: Access class. One of the authenticated classes of the default access policy prefixed with the string <emphasis>onvif:</emphasis>, i.e. "onvif:Administrator", "onvif:Operator" or "onvif:User" as defined also within the ONVIF Core Specification. This access level will be used to authorize access to the required function.</para>
+          </listitem>
+        </itemizedlist>
+      </para>
+      <para>The signature is evaluated by applying ECDSA using secp256r1 and SHA-256 hashing to the base64-encoded header and payload, concatenated by a '.'</para>
+      <programlisting>ECDSA-SHA256 (base64UrlEncode(header) + "." +
+              base64UrlEncode(payload),
+              secp256r1))</programlisting>
+      <para>The evaluated signature is further appended to the encoded header and payload with a '.' separator, to get the final JWT. Appendix <xref linkend="_Ref460769599"/> demonstrates the construction or a JWT.</para>
+      <para>The signature is used to verify the message wasn't changed along the way, and, in the case of tokens signed with a private key, it can also verify the identity of who signed the JWT.</para>
+      <section>
+        <title>Usage of JWT-based client authentication over HTTP</title>
+        <para>Since authentication tokens contain sensitive information that could be easily used to perform replay attacks, JWT-based client authentication shall not be used over unencrypted HTTP connections.</para>
```

Comment:
Shall a device reject a HTTP request, or is this a client requirement? Similar sentence is in the core spec.